feat: learn the date object by building a date formatter (#50504)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-the-date-object-by-building-a-date-formatter/646e44ec9bd212021dc7de44.md

---
id: 646e44ec9bd212021dc7de44
title: Step 46
challengeType: 0
dashedName: step-46
---

# --description--

> You will add the `reverse()` method to the end of `split()` method.

Reviewer: This is also the first time we are introducing the `reverse` method so we should provide a brief explanation here.

```suggestion
The `reverse()` method is used to reverse an array in place. 

Chain the `reverse()` method to the end of `split()` method.
```

Author: added

  justify-content: center;
  gap: 8px;
}

.title {
  font-size: 2.5rem;
}

.date-svg{
  width: 30px;
  height: 30px;
}

.divider {
  margin: auto;
  width: 150px;
  height: 10px;
  background-color: var(--yellow);
}

.dropdown-container {
  width: 100%;
  margin: 40px auto;
  position: relative;
}

select {
  display: block;
  margin: 20px auto;
  height: 50px;
  font-size: 100%;
  font-weight: bold;

  cursor: pointer;
  background-color: var(--white);
  border: none;
  color: var(--dark-grey);
  appearance: none;
  padding: 10px 38px 10px 10px;
}

.select-icon{
  position: absolute;
  top: 4px;
  right: 4px;
  pointer-events: none;
  width: 30px;
  height: 36px;
  padding-left: 4px;
}

#current-date{
  font-size: 2rem;
  text-align:center;
}
```

```js
const currentDateParagraph = document.getElementById("current-date");
const dateOptionsSelectElement = document.getElementById("date-options");

const date = new Date();
const day = date.getDate();
const month = date.getMonth() + 1;
const year = date.getFullYear();
const hours = date.getHours();
const minutes = date.getMinutes();

const formattedDate = `${day}-${month}-${year}`;
currentDateParagraph.textContent = formattedDate;

> dateOptionsSelectElement.addEventListener("change", () => {
>   switch(dateOptionsSelectElement.value) {
>     case 'yyyy-mm-dd':
> --fcc-editable-region--
>       currentDateParagraph.textContent = formattedDate
>       .split("-")
> 
> --fcc-editable-region--
>     break;
>     default: 
>       currentDateParagraph.textContent = formattedDate
>     break
>   }
> })

Reviewer 2: ```suggestion
dateOptionsSelectElement.addEventListener("change", () => {
  switch(dateOptionsSelectElement.value) {
    case 'yyyy-mm-dd':
--fcc-editable-region--
      currentDateParagraph.textContent = formattedDate.split("-");
--fcc-editable-region--
      break;
    default: 
      currentDateParagraph.textContent = formattedDate;
  }
});
```

      </div>
      <p id="current-date"></p>
    </main>
  </body>
  <script src="script.js"></script>
</html>
```

```css
*{
  margin: 0;
  padding: 0;
  box-sizing: border-box
}
:root {
  --dark-grey: #0a0a23;
  --white: #ffffff;
  --yellow: #f1be32;
}

body{
  background-color: var(--dark-grey);
  color: var(--white);
}

.title-container{
  margin: 10px 0;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
}

.title {
  font-size: 2.5rem;
}

.date-svg{
  width: 30px;
  height: 30px;
}

.divider {
  margin: auto;
  width: 150px;
  height: 10px;
  background-color: var(--yellow);
}

.dropdown-container {
  width: 100%;
  margin: 40px auto;
  position: relative;
}

select {
  display: block;
  margin: 20px auto;
  height: 50px;
  font-size: 100%;
  font-weight: bold;

  cursor: pointer;
  background-color: var(--white);
  border: none;
  color: var(--dark-grey);
  appearance: none;
  padding: 10px 38px 10px 10px;
}

.select-icon{
  position: absolute;
  top: 4px;
  right: 4px;
  pointer-events: none;
  width: 30px;
  height: 36px;
  padding-left: 4px;
}

#current-date{
  font-size: 2rem;
  text-align:center
}
```

```js

> const currentDateParagraph = document.getElementById("current-date")
> const dateOptionsSelectElement = document.getElementById("date-options")

Reviewer: ```suggestion
const currentDateParagraph = document.getElementById("current-date");
const dateOptionsSelectElement = document.getElementById("date-options");
```

const date = new Date();
const day = date.getDate();
const month = date.getMonth() + 1;
const year = date.getFullYear();
const hours = date.getHours();
const minutes = date.getMinutes();

> const formattedDate = `${day}-${month}-${year}`
> currentDateParagraph.textContent = formattedDate

Reviewer: ```suggestion
const formattedDate = `${day}-${month}-${year}`;
currentDateParagraph.textContent = formattedDate;
```

> dateOptionsSelectElement.addEventListener("change", ()=>{

Reviewer: ```suggestion
dateOptionsSelectElement.addEventListener("change", () => {
```

>   switch(dateOptionsSelectElement.value){

Reviewer: ```suggestion
  switch(dateOptionsSelectElement.value) {
```

    case 'yyyy-mm-dd':

>       currentDateParagraph.textContent = formattedDate
>       .split("-")
> --fcc-editable-region--

Reviewer 3: ```suggestion
--fcc-editable-region--
      currentDateParagraph.textContent = formattedDate
      .split("-")
```
Same as before
